File: String/AddBinary.js
/*
Given two binary strings, return their sum (also a binary string).

For example,
a = "11"
b = "1"
Return "100".
*/

/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function(a, b) {
    var i = a.length - 1;
    var j = b.length - 1;
    var carry = 0;
    var res = "";
    while(i >= 0 || j >= 0 || carry > 0){
        var n1 = 0;
        var n2 = 0;
        if(i >= 0) n1 = parseInt(a.charAt(i));
        if(j >= 0) n2 = parseInt(b.charAt(j));
        var cur = n1 + n2 + carry;
        res = cur % 2 + res;
        carry = Math.floor(cur / 2);
        i--;
        j--;
    }
    return res;
};

File: String/AddBinary.java
/*
Given two binary strings, return their sum (also a binary string).

For example,
a = "11"
b = "1"
Return "100".
*/

public class Solution {
    public String addBinary(String a, String b) {
        StringBuilder res = new StringBuilder();
        int carry = 0;
        int i = a.length() - 1;
        int j = b.length() - 1;
        while(i >= 0 || j >= 0 || carry > 0){
            int num1 = 0;
            int num2 = 0;
            if(i >= 0) num1 = a.charAt(i) - '0';
            if(j >= 0) num2 = b.charAt(j) - '0';
            int n = num1 + num2 + carry;
            res.insert(0, (n % 2));
            carry = n / 2;
            i--;
            j--;
        }
        return res.toString();
    }
}